drivers/console/Kconfig

```diff
@@ -291,6 +291,13 @@ config NATIVE_POSIX_CONSOLE_INIT_PRIORITY
 	help
 	  Device driver initialization priority.
 
+config SEMIHOST_CONSOLE
+	bool "Use semihosting for console"
+	select CONSOLE_HAS_DRIVER
+	depends on CPU_CORTEX
```

Comment:
Since the semihosting trap mechanism used here is specific to Cortex-M, it should `depends on CPU_CORTEX_M`.